buildSrc/src/main/java/net/neoforged/neodev/jcc/DownloadInstaller.java

```diff
@@ -0,0 +1,55 @@
+package net.neoforged.neodev.jcc;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+/**
+ * Downloads an arbitrary NeoForge installer by version.
+ * <p>Sometimes we cannot use Gradle configurations since Gradle would replace references to net.neoforged:neoforge
+ * with a reference to the current project. That prevents downloading different versions.
+ */
+public abstract class DownloadInstaller extends DefaultTask {
```

Comment:
As generic as this is, I would name it `DownloadFile`. It has no code that's specific to downloading installers, it doesn't even build installer urls by itself.